File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-2.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number[][]}
     */
    levelOrder(root) {
        let que = [];
        let res = [];
        if (root) {
            que.push(root);
        }
        while (que.length > 0) {
            let length = que.length;
            let layer = [];
            for (let i = 0 ;  i < length; i++){
                let curr = que.shift();
                if (curr.left) {
                    que.push(curr.left);
                }
                if (curr.right) {
                    que.push(curr.right);
                }
                layer.push(curr.val)
            }
            res.push(layer)
        }
        return res
    }
}

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        # iteration
        curr = root
        stack = []
        ans = []
        while curr or stack:
            # if len(ans) == k:
            #     return ans[k - 1]
            while curr:
                stack.append(curr)
                curr = curr.left
            curr = stack.pop()
            ans.append(curr.val)
            curr = curr.right
        return ans[k-1]